modules/autocomplete-feed/main/index.coffee

```diff
@@ -0,0 +1,104 @@
+sortItems = (valueLookup) ->
+  valueLookup ?= hx.identity
+  (a, b) ->
+    if not a.disabled and b.disabled then -1
+    else if a.disabled and not b.disabled then 1
+    else hx.sort.compare(valueLookup(a), valueLookup(b))
+
+trimTrailingSpaces = (term) ->
+  newTerm = term
+  while newTerm.lastIndexOf(' ') is newTerm.length - 1
+    newTerm = newTerm.slice(0, newTerm.length - 1)
+  newTerm
+
+class AutocompleteFeed
+  constructor: (options = {}) ->
+    self = this
+
+    defaults =
+      filter: undefined
+      filterOptions: undefined
+      matchType: 'contains'
+      showOtherResults: false
+      trimTrailingSpaces: false
+      valueLookup: undefined
+
+    if options.valueLookup?
+      # default searchValue if valueLookup is defined
+      defaults.filterOptions =
+        searchValues: (datum) -> [self._.options.valueLookup(datum)]
+
+    resolvedOptions = hx.merge.defined defaults, options
+
+    # defined here so we can use the resolved options
+    resolvedOptions.filter ?= (items, term) ->
+      hx.filter[resolvedOptions.matchType](items, term, resolvedOptions.filterOptions)
+        .sort sortItems(resolvedOptions.valueLookup)
+
+    @_ =
+      options: resolvedOptions
+      resultsCache: new hx.Map
+
+
+  clearCache: ->
+    @_.resultsCache = new hx.Map
+    this
+
+  filter: (term = '', callback) ->
+    _ = @_
+    thisFilter = term + hx.randomId()
+    _.lastFilter = thisFilter
```

Comment:
By 'this' I mean the 'only callback with the last called filter'.
Everything is run and cached for every filter but the callback is only called when `thisFilter === _.lastFilter`